plugins/services/src/js/components/PodVolumeTable.js

```diff
@@ -0,0 +1,204 @@
+    return classNames({
+      active: prop === sortBy.prop,
+      clickable: row == null,
+      "text-overflow": prop === "profile"
+    });
+  }
+
+  getColumnHeading(prop, order, sortBy) {
+    const caretClassNames = classNames("caret", {
+      [`caret--${order}`]: order != null,
+      "caret--visible": prop === sortBy.prop
+    });
+
+    const headingStrings = {
+      id: "ID",
+      host: "Host",
+      type: "Volume Type",
+      profile: "Volume Profile",
+      name: "Volume Name",
+      size: "Size",
+      status: "Status"
+    };
+
+    return (
+      <span>
+        {headingStrings[prop]}
+        <span className={caretClassNames} />
+      </span>
+    );
+  }
+
+  getColumns() {
+    return [
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: "id",
+        render: this.renderIDColumn,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: "host",
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: "type",
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: "profile",
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: "name",
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: "size",
+        render: this.renderSizeColumn,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: "status",
+        render: this.renderStatusColumn,
+        sortable: true
+      }
+    ];
+  }
+
+  renderIDColumn(prop, row) {
+    const id = row[prop];
+    const { nodeID, taskID } = this.props.params;
+    const volumeID = encodeURIComponent(id);
+    const serviceID = encodeURIComponent(this.props.params.id);
+    const currentroutePath = reconstructPathFromRoutes(this.props.routes);
+    let routePath = null;
+
+    if (currentroutePath === "/services/detail/:id/podvolumes") {
+      routePath = `/services/detail/${serviceID}/podvolumes/${volumeID}`;
+    } else if (
+      currentroutePath === "/services/detail/:id/tasks/:taskID/podvolumes"
+    ) {
+      routePath = `/services/detail/${serviceID}/tasks/${taskID}/podvolumes/${volumeID}`;
+    } else if (currentroutePath === "/nodes/:nodeID/tasks/:taskID/podvolumes") {
+      routePath = `/nodes/${nodeID}/tasks/${taskID}/podvolumes/${volumeID}`;
+    }
+
+    return <Link className="table-cell-link-primary" to={routePath}>{id}</Link>;
+  }
+
+  renderSizeColumn(prop, row) {
+    const { size } = row;
+    let unit = "MiB";
+
+    if (
+      row.type !== VolumeDefinitions.PERSISTENT.type &&
```

Comment:
I thought only `persistent` and `dss` will be shown anyway. Though I can't run a pod with DSS volume, but that's a different story :)